src/utils/pickupTimeEstimator.ts

```diff
@@ -200,27 +233,128 @@ export const addOnsiteTravelDuration = async (serviceTime, locationId) => {
   return addDuration(serviceTime, onsiteTravelDuration)
 }
 
-/**
- *  Returns true if the named location has a special delivery schedule (e.g. is
- *  a scholar room)
- *
- *  TODO: Implement
- */
-export const hasSpecialDeliverySchedule = (locationId: string) => {
-  // Return true if locationId matches room known to have a specific delivery schedule
-  return false
+type AdjustToSpecialSchedule = {
+  hasSpecialDeliverySchedule: boolean
+  arrivalAtHoldshelf: string
+  rationale: RationaleEntry[]
 }
 
 /**
  *  Given a location id and a timestamp string, returns a new timestamp string
  *  representing the future delivery time for the location given the known
  *  delivery schedule.
- *
- *  TODO: Implement
  */
-export const adjustToSpecialSchedule = (locationId, time): string => {
-  // Update time to next delivery schedule after given time
-  return time
+export const adjustToSpecialSchedule = (
+  locationId,
+  time
+): AdjustToSpecialSchedule => {
+  let hasSpecialDeliverySchedule = false
+  const adjustedSpecialScheduleTime = new Date(time)
+  const secondFloorScholarRooms = ["mal17", "mala", "malc", "maln", "malw"]
+  const mapRooms = ["mapp8", "mapp9", "map08"]
+  let getFirstHour
+  let getNextHour
+  let getLastHour
+  // Subtract 1ms to coerce T10:00.000 to be evaluated as T09:59.999
+  // so that the evaluated next delivery for something avail at 10am sharp is 10am
+  const specialFudgeMs = -1
+  const offsetMs =
+    1000 *
+      60 *
+      (adjustedSpecialScheduleTime.getTimezoneOffset() - 60 * nyOffset()) -
+    adjustedSpecialScheduleTime.getMilliseconds() +
+    specialFudgeMs
+
+  // adjust time to simulate being in New York
+  adjustedSpecialScheduleTime.setTime(
+    adjustedSpecialScheduleTime.getTime() + offsetMs
+  )
+
+  const loggableTimestamp = (date) => {
+    return new Date(date.getTime() - offsetMs + specialFudgeMs).toISOString()
+  }
+
+  const rationale = []
+
+  if (secondFloorScholarRooms.includes(locationId)) {
+    hasSpecialDeliverySchedule = true
+    getFirstHour = (time) => {
+      const day = time.getDay()
+      return day === 0 ? 14 : 10
```

Comment:
I think it might make sense to have a constant that maps day keys to labels just to make things more legible. I also wouldn't mind having closing times in named constants either.

Something like

const DAYS = {
   MONDAY: 0
   TUESDAY: 1
}
etc